mmeval/metrics/rouge.py

```diff
@@ -0,0 +1,327 @@
+            Defaults to None.
+        accumulate (str): Useful in case of multi-reference rouge score.
+            ``avg`` takes the avg of all references with respect to predictions
+            ``best`` takes the best fmeasure score obtained between prediction
+            and multiple corresponding references.
+            Defaults to ``best``.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+        >>> from mmeval import ROUGE
+        >>> predictions = ['the cat is on the mat']
+        >>> references = [['a cat is on the mat']]
+        >>> metric = ROUGE(rouge_keys='L')
+        >>> metric.add(predictions, references)
+        >>> results = metric.compute_metric()
+        >>> {'rougeL_fmeasure': 0.8333333,'rougeL_precision': 0.8333333,'rougeL_recall': 0.8333333}  # noqa: E501
+    """
+
+    def __init__(
+        self,
+        rouge_keys: Union[List, Tuple, int, str] = (1, 2, 'L'),
+        use_stemmer: bool = False,
+        normalizer: Optional[Callable] = None,
+        tokenizer: Union[Callable, str, None] = None,
+        accumulate: str = 'best',
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+        assert accumulate in [
+            'best', 'avg'
+        ], (f'Wrong accumulate {accumulate}. Supported accumulate are '
+            '"best" and "avg"')
+        if isinstance(rouge_keys, int) or isinstance(rouge_keys, str):
+            rouge_keys = [rouge_keys]
+        # Check the legitimacy of the rouge_keys
+        for rouge_key in rouge_keys:
+            if isinstance(rouge_key, int):
+                if rouge_key < 1 or rouge_key > 9:
+                    raise ValueError(
+                        f'Got unknown rouge key {rouge_key}. Expected to be one of {1 - 9} or L'  # noqa: E501
+                    )
+            elif rouge_key != 'L':
+                raise ValueError(
+                    f'Got unknown rouge key {rouge_key}. Expected to be one of {1 - 9} or L'  # noqa: E501
+                )
+        self.rouge_keys = rouge_keys
+
+        if use_stemmer:
+            if try_import('nltk'):
+                import nltk
+                self.stemmer = nltk.stem.porter.PorterStemmer()
+            else:
+                raise ValueError(
+                    'The nltk package is needed to use stemmer, check https://www.nltk.org/install.html for installation.'  # noqa: E501
+                )
+        else:
+            self.stemmer = None
+        self.normalizer = normalizer
+
+        # Select tokenizer according to the entered value.
+        self.tokenizer_fn = None
+        if callable(tokenizer):
+            self.tokenizer_fn = tokenizer
+        elif isinstance(tokenizer, str):
+            self.tokenizer_fn = _get_tokenizer(tokenizer)
+            if self.tokenizer_fn is None:
+                raise ValueError(
+                    "Right now, `tokenizer` only supports pre-defined 'en' or 'cn'."  # noqa: E501
+                )
```

Comment:
Code style suggestion 
```suggestion
                raise ValueError("Right now, `tokenizer` only supports "
                                 "pre-defined 'en' or 'cn'.")
```